internal/planner/configuration_test.go

```diff
@@ -0,0 +1,192 @@
+			Namespace: "smbshares",
+			UID:       "phonyuid3",
+		},
+		Spec: sambaoperatorv1alpha1.SmbShareSpec{
+			ShareName:      "",
+			ReadOnly:       false,
+			Browseable:     true,
+			SecurityConfig: "sc1",
+			CommonConfig:   "",
+			Storage: sambaoperatorv1alpha1.SmbShareStorageSpec{
+				Pvc: &sambaoperatorv1alpha1.SmbSharePvcSpec{
+					Name: "mydata",
+					Path: "share3",
+				},
+			},
+		},
+	}
+}
+
+func sampleADSecConfig1() *sambaoperatorv1alpha1.SmbSecurityConfig {
+	return &sambaoperatorv1alpha1.SmbSecurityConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sc1",
+			Namespace: "smbshares",
+			UID:       "phonyadscuid1",
+		},
+		Spec: sambaoperatorv1alpha1.SmbSecurityConfigSpec{
+			Mode:  "active-directory",
+			Realm: "FOO.TEST",
+		},
+	}
+}
+
+func testSimpleUpdate(t *testing.T, state *smbcc.SambaContainerConfig) {
+	assert.Len(t, state.Shares, 0)
+	assert.Len(t, state.Configs, 0)
+	assert.Len(t, state.Globals, 0)
+
+	p := New(InstanceConfiguration{
+		SmbShare:     sampleSmbShare1(),
+		GlobalConfig: &conf.OperatorConfig{},
+	}, state)
+
+	// apply changes
+	changed, err := p.Update()
+	assert.NoError(t, err)
+	assert.True(t, changed)
+
+	// changes already applied
+	changed, err = p.Update()
+	assert.NoError(t, err)
+	assert.False(t, changed)
+
+	assert.Len(t, state.Shares, 1)
+	assert.Len(t, state.Configs, 1)
+	assert.Len(t, state.Globals, 1)
+	assert.Contains(t, state.Shares, smbcc.Key("share1"))
+}
+
+func testSecondShare(t *testing.T, state *smbcc.SambaContainerConfig) {
+	assert.Len(t, state.Shares, 0)
+	assert.Len(t, state.Configs, 0)
+	assert.Len(t, state.Globals, 0)
+
+	p := New(InstanceConfiguration{
+		SmbShare:     sampleSmbShare1(),
+		GlobalConfig: &conf.OperatorConfig{},
+	}, state)
+
+	// apply changes
+	changed, err := p.Update()
+	assert.NoError(t, err)
+	assert.True(t, changed)
+
+	p2 := New(InstanceConfiguration{
+		SmbShare:     sampleSmbShare2(),
+		GlobalConfig: &conf.OperatorConfig{},
+	}, state)
+	changed, err = p2.Update()
+	assert.NoError(t, err)
+	assert.True(t, changed)
+
+	assert.Len(t, state.Shares, 2)
+	assert.Len(t, state.Configs, 1)
+	assert.Len(t, state.Globals, 1)
+	assert.Contains(t, state.Shares, smbcc.Key("share1"))
+	assert.Contains(t, state.Shares, smbcc.Key("share2"))
+	assert.Contains(t, state.Configs[p.instanceID()].Shares, smbcc.Key("share1"))
+	assert.Contains(t, state.Configs[p.instanceID()].Shares, smbcc.Key("share2"))
+}
+
+func testADShare(t *testing.T, state *smbcc.SambaContainerConfig) {
+	assert.Len(t, state.Shares, 0)
+	assert.Len(t, state.Configs, 0)
+	assert.Len(t, state.Globals, 0)
+
+	p := New(InstanceConfiguration{
+		SmbShare:       sampleSmbShare3(),
+		SecurityConfig: sampleADSecConfig1(),
+		GlobalConfig:   &conf.OperatorConfig{},
+	}, state)
+
+	// apply changes
+	changed, err := p.Update()
+	assert.NoError(t, err)
+	assert.True(t, changed)
+
+	assert.Len(t, state.Shares, 1)
+	assert.Len(t, state.Configs, 1)
+	assert.Len(t, state.Globals, 2)
```

Comment:
I had to go through the _Update()_ func to realize that `Realm` from `SmbSecurityConfigSpec` was the one getting added as the next(2nd) key in `Globals` map.